netdata-installer.sh

```diff
@@ -628,6 +634,80 @@ bundle_libwebsockets
 
 # -----------------------------------------------------------------------------
 
+build_judy() {
+  local env_cmd=''
+
+  if [ -z "${DONT_SCRUB_CFLAGS_EVEN_THOUGH_IT_MAY_BREAK_THINGS}" ]; then
+    env_cmd="env CFLAGS= CXXFLAGS= LDFLAGS="
+  fi
+
+  pushd "${1}" > /dev/null || exit 1
+  run ${env_cmd} libtoolize --force --copy
+  run ${env_cmd} aclocal
+  run ${env_cmd} autoheader
+  run ${env_cmd} automake --add-missing --force --copy --include-deps
+  run ${env_cmd} autoconf
+  run ${env_cmd} ./configure
+  run ${env_cmd} make -C src
+  run ${env_cmd} ar -r src/libJudy.a src/Judy*/*.o
+  popd > /dev/null || exit 1
+}
+
+copy_judy() {
+  target_dir="${PWD}/externaldeps/libjudy"
+
+  run mkdir -p "${target_dir}" || return 1
+
+  run cp "${1}/src/libJudy.a" "${target_dir}/libJudy.a" || return 1
+  run cp "${1}/src/Judy.h" "${target_dir}/Judy.h" || return 1
+}
+
+bundle_judy() {
+  # If --build-judy flag or no Judy on the system and we're building the dbengine, bundle our own libJudy.
+  if [ -n "${NETDATA_DISABLE_DBENGINE}" ] && [ -z "${NETDATA_BUILD_JUDY}" ] && [ -e /usr/include/Judy.h ] ; then
+    return 0
+  fi
+
+  progress "Prepare libJudy"
+
+  JUDY_PACKAGE_VERSION="$(cat packaging/judy.version)"
+
+  tmp="$(mktemp -d -t netdata-judy-XXXXXX)"
+  JUDY_PACKAGE_BASENAME="v${JUDY_PACKAGE_VERSION}.tar.gz"
+
+  if fetch_and_verify "judy" \
+    "https://github.com/netdata/libjudy/archive/${JUDY_PACKAGE_BASENAME}" \
+    "${JUDY_PACKAGE_BASENAME}" \
+    "${tmp}" \
+    "${NETDATA_LOCAL_TARBALL_OVERRIDE_JUDY}"; then
+    if run tar -xf "${tmp}/${JUDY_PACKAGE_BASENAME}" -C "${tmp}" &&
+      build_judy "${tmp}/libjudy-${JUDY_PACKAGE_VERSION}" &&
+      copy_judy "${tmp}/libjudy-${JUDY_PACKAGE_VERSION}" &&
+      rm -rf "${tmp}"; then
+      run_ok "libJudy built and prepared."
+      NETDATA_CONFIGURE_OPTIONS="${NETDATA_CONFIGURE_OPTIONS} --with-libjudy=externaldeps/libJudy"
+    else
+      run_failed "Failed to build libJudy."
+      if [ -n "${NETDATA_REQUIRE_CLOUD}" ]; then
+        exit 1
+      else
+        defer_error_highlighted "Failed to build libJudy. dbengine support will be disabled."
```

Comment:
Is this accurate here? Failure to build does not imply disabling dbengine since the system library will be used. And in the same spirit, why do we give priority to building the bundle over the system Judy?

Reply:
We don't preferentially build Judy instead ofusing the system copy. The check that decides on this is the first thing in this function.

So, if we actually get to this point and the build has failed, we don't have Judy at all, so we can't build with the dbengine.

Reply:
The one caveat to that though is if the user specifically tells us to build libJudy with the `--build-judy` flag, in that case we just ignore any existing copy and only try to use the built version, but we should probably bail if that flag was passed and the build failed.

Reply:
well it could go there in case that the system header is in different path (e.g. we detected system judy is missing as mistake)?